### PR TITLE
loosen the restriction of output_size in conv2d_transpose

### PR DESCRIPTION
**Pull request by cjld**
output_size of conv2d_transpose has a restriction and causes that it is impossible to upsample a 7x7 input tensor to a 14x14 output tensor with a 5x5 kernel. this PR will loosen the restriction.

A example:
```python

import paddle
import paddle.fluid as fluid
import numpy as np



xx = np.array(range(7*7)).astype(np.float32).reshape((1,7,7,1))+1
ww = np.zeros((5,5,25,1)).astype(np.float32)
for i in range(5):
    for j in range(5):
        ww[i,j,i*5+j,0] = 1

exe = fluid.Executor(fluid.CPUPlace())

startup_program = fluid.Program()
program = fluid.Program()
nw = ww.transpose((3,2,0,1))
with fluid.program_guard(program, startup_program):
    blk = program.global_block()
    startup_program.global_block().create_parameter(
        dtype='float32', shape=nw.shape, name='w')
        #, with_initializer=True, initializer=fluid.initializer.Constant(1))
    var = blk.create_parameter(dtype='float32', shape=nw.shape, name='w')
    op = blk.append_op(
        type='assign_value',
        outputs={'Out': [var]},
        attrs={
            'dtype': var.dtype,
            'shape': list(var.shape),
            'fp32_values': map(lambda x:float(x), ww.flatten())
    })
    shape = [1,7,7]
    img = fluid.layers.data(name='img', shape=shape, dtype='float32')

    y = blk.create_var(name = 'y', dtype='float32')
    blk.append_op(
        type='conv2d_transpose',
        inputs={'Input': [img],
                'Filter': [var]},
        outputs={'Output': y},
        attrs={
            'output_size': [14,14],
            'strides': [2,2],
            'paddings': [2,2],
            'dilations': [1,1],
            'groups': 1,
            'use_cudnn': False
        })

exe.run(startup_program)
result = exe.run(program, feed={'img':xx.transpose(0,3,1,2)}, fetch_list={y, var})

print result[1][0,0]
print result[1][0,24]
```

results:
```
[[ 9.  0. 10.  0. 11.  0. 12.  0. 13.  0. 14.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [16.  0. 17.  0. 18.  0. 19.  0. 20.  0. 21.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [23.  0. 24.  0. 25.  0. 26.  0. 27.  0. 28.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [30.  0. 31.  0. 32.  0. 33.  0. 34.  0. 35.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [37.  0. 38.  0. 39.  0. 40.  0. 41.  0. 42.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [44.  0. 45.  0. 46.  0. 47.  0. 48.  0. 49.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]]

[[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0.  1.  0.  2.  0.  3.  0.  4.  0.  5.  0.  6.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0.  8.  0.  9.  0. 10.  0. 11.  0. 12.  0. 13.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0. 15.  0. 16.  0. 17.  0. 18.  0. 19.  0. 20.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0. 22.  0. 23.  0. 24.  0. 25.  0. 26.  0. 27.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0. 29.  0. 30.  0. 31.  0. 32.  0. 33.  0. 34.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
 [ 0.  0. 36.  0. 37.  0. 38.  0. 39.  0. 40.  0. 41.  0.]
 [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]]
```

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -2302,16 +2302,17 @@ def conv2d_transpose(input,
...
+           W_{out} \in [ W^\prime_{out}, W^\prime_{out} + strides[1] )
 
     Args:
         input(Variable): The input image with [N, C, H, W] format.
         num_filters(int): The number of the filter. It is as same as the output
             image channel.
         output_size(int|tuple|None): The output image size. If output size is a
-            tuple, it must contain two integers, (image_H, image_W). This
-            parameter only works when filter_size is None.
+            tuple, it must contain two integers, (image_H, image_W).
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 这里的设置有好几种情况:
1. 设置filter_size, padding, stride，不设置output_size，会计算输出的shape:output_size
2. 设置output_size, padding,stride，会自动计算filter_size，输出的shape是output_size
3. 也有同时设置filter_size, padding, stride,output_size这种情况？ 

这些都应该在上面doc里说明白吧，麻烦帮忙改进下上面文档。

**Review by qingqing01 — COMMENTED** [review]
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/unittests/test_conv2d_transpose_op.py

https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/unittests/test_conv3d_transpose_op.py

这里面记得加单测

**Comment by cjld** [comment]
test results:
```
 test 137
    Start 137: test_conv2d_transpose_op

137: Test command: /usr/bin/env "PYTHONPATH=/paddle/build/python" "/opt/python/cp27-cp27mu/bin/python" "/paddle/tools/test_runner.py" "test_conv2d_transpose_op"
137: Test timeout computed to be: 9.99988e+06
1/1 Test #137: test_conv2d_transpose_op .........   Passed    4.31 sec

The following tests passed:
        test_conv2d_transpose_op

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   4.33 sec

```

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Pull request merged**